Cache autoupdate check to run only once per 24 hours

murataslan1 opened the pull request:
## Summary
This PR caches the autoupdate check so it only runs once per 24 hours instead of on every application start.

## Problem
When running drawio frequently or in scripts (even with `--help` flag), the autoupdate check runs every time, outputting messages like:
```
Checking for beta autoupdate feature for deb/rpm distributions
Found package-type: deb
```

This is especially annoying when running drawio in a loop or automated scripts.

## Solution
- Store the last update check timestamp in electron-store
- Only perform update check if 24 hours have passed since the last check
- Existing behavior is preserved: users can still manually check for updates via the menu

## Changes
- Modified `src/main/electron.js` to cache update check timestamp

Fixes #2262

murataslan1 committed:
Cache autoupdate check to run only once per 24 hours

Previously, the update check ran on every application start, which was
annoying when running drawio in scripts or frequently from CLI (even
with --help flag).

This change caches the last update check timestamp and only performs
a new check if 24 hours have passed since the last one.

Fixes #2262

andrei-melnikau-plateau commented on src/main/electron.js:
```diff
@@ -1171,8 +1171,18 @@ app.whenReady().then(() =>
 	
-	if (store == null || (!disableUpdate && !store.get('dontCheckUpdates')))
+	// Cache update check - only check once per 24 hours to avoid repeated messages
+	const UPDATE_CHECK_INTERVAL = 24 * 60 * 60 * 1000; // 24 hours
```
Please make this UPDATE_CHECK_INTERVAL configurable. Hardcoding 24h looks like a kludge. What if I want to check once per week?

alexanderLinear replied:
to me a command line option for suppressing update checks all together, might be helpful for real scripting. some automated build environments aren't permitted for the internet and thus a network access timeout might not be desirable.

m-mohamedin replied:
@alexanderLinear Running the command with '--disable-update' or setting environment variable DRAWIO_DISABLE_UPDATE=true would stop checking for updates

murataslan1 committed:
Make update check interval configurable

Allow users to configure the update check interval via 'updateCheckIntervalHours'
setting in electron-store. Default remains 24 hours for backward compatibility.

Example: Set to 168 for weekly checks.

codeCraft-Ritik reviewed (commented):
Nice

murataslan1 commented:
Hi there!  Just a friendly follow-up on this PR. Is there anything I can do to help move this forward? Happy to address any feedback or make adjustments if needed. Thanks for your time!
